board/libc.h

```diff
@@ -6,32 +6,36 @@ void delay(int a) {
 }
 
 void *memset(void *str, int c, unsigned int n) {
-  unsigned int i;
-  for (i = 0; i < n; i++) {
-    *((uint8_t*)str) = c;
-    ++str;
+  uint8_t *s = str;
+  for (unsigned int i = 0; i < n; i++) {
+    *s = c;
+    s++;
   }
   return str;
```

Comment:
To keep it the same, I should return `s` instead of `str`, but this is a more canonical return for memset I think.

Reply:
Fine to change